dipy/direction/bootstrap_direction_getter.py

```diff
@@ -0,0 +1,130 @@
+from warnings import warn
+import numpy as np
+
+from dipy.reconst import shm
+from dipy.core.geometry import cart2sphere
+from dipy.tracking.local.interpolation import trilinear_interpolate4d
+
+from dipy.data import default_sphere
+from dipy.direction.closest_peak import BaseDirectionGetter
+
+
+default_SH = 4
```

Comment:
Do we really this global variable ? you use it only once

Reply:
@MrBago can you comment on this? Thanks!

Reply:
The benefit here is marginal, but this is a common pattern (https://refactoring.guru/replace-magic-number-with-symbolic-constant) and I'd recommend we leave it.

Reply:
Ok, and Thanks for this link @MrBago !

Reply:
Are you assuming that this class will work only with shm.py models? Or you are supporting other models here?

Also ``default_SH`` should be ``DEFAULT_SH`` but why 4? Isn't this a bit low?

Reply:
@MrBago, could you comment on the model support question?

Regarding DEFAULT_SH: I think it's because of this "The RMS error of the noise estimates show a minimum at SH order 4 for b-values 1000 and 2000, and a minimum at SH order 6 for b-value of 3000. After the minimum, the RMS error increases with SH order.", from @MrBago's dissertation.

Reply:
This BootstrapDirectionGetter will work with any model, but it implements the SH bootstrap. DTI bootstrap was published quite a few years ago, but it doesn't work as well as the SH bootstrap. If there is interest in adding a DTI bootstrap class, I think it should be an independent effort because the underlying implementation will need to be substantially different.

Reply:
Thanks for this explanation @kesshijordan @MrBago!

Can you just switch to uppercase this magic variable @kesshijordan ?